src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,89 @@
+package edu.ksu.canvas.attendance.submitter;
+
+
+import edu.ksu.canvas.attendance.entity.AttendanceAssignment;
+import edu.ksu.canvas.attendance.form.CourseConfigurationForm;
+import edu.ksu.canvas.attendance.services.CanvasApiWrapperService;
+import edu.ksu.canvas.exception.ObjectNotFoundException;
+import edu.ksu.canvas.model.assignment.Assignment;
+import edu.ksu.canvas.oauth.OauthToken;
+import org.apache.log4j.Logger;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Optional;
+
+@Component
+@Scope(value="session")
+public class AssignmentValidator {
+
+    private static final Logger LOG = Logger.getLogger(AssignmentValidator.class);
+
+    public Error validateAttendanceAssignment(Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) {
```

Comment:
Error should probably not be used in this manner. This Error class is in java.lang and is intended for errors that the application cannot recover from. Actually even if that wasn't the case, a class which extends Throwable should used for throwing things and not returning values.

Throwable and classes that extend them can be used by the throws keyword. For example, Exception extends Throwable. Throwables can also be caught, but since most of the time catch statements catch Exception, they are not often caught.